pkg/controller/atlascluster/cluster.go

```diff
@@ -84,22 +84,48 @@ func (r *AtlasClusterReconciler) ensureClusterState(ctx *workflow.Context, proje
 	}
 }
 
-// clusterMatchesSpec will merge everything from the Spec into existing Cluster and use that to detect change.
-// Direct comparison is not feasible because Atlas will set a lot of fields to default values, so we need to apply our changes on top of that.
-func clusterMatchesSpec(log *zap.SugaredLogger, cluster *mongodbatlas.Cluster, spec mdbv1.AtlasClusterSpec) (bool, error) {
-	clusterMerged := mongodbatlas.Cluster{}
-	if err := compat.JSONCopy(&clusterMerged, cluster); err != nil {
-		return false, err
+func cleanupCluster(cluster mongodbatlas.Cluster) mongodbatlas.Cluster {
+	cluster.ID = ""
+	cluster.MongoDBVersion = ""
+	cluster.MongoURI = ""
+	cluster.MongoURIUpdated = ""
+	cluster.MongoURIWithOptions = ""
+	cluster.SrvAddress = ""
+	cluster.StateName = ""
+	cluster.ReplicationFactor = nil
+	cluster.ReplicationSpec = nil
+	cluster.ConnectionStrings = nil
+	return cluster
+}
+
+// mergedCluster will return the result of merging AtlasClusterSpec with Atlas Cluster
+func mergedCluster(cluster mongodbatlas.Cluster, spec mdbv1.AtlasClusterSpec) (result mongodbatlas.Cluster, err error) {
+	if err = compat.JSONCopy(&result, cluster); err != nil {
+		return
+	}
+
+	if err = compat.JSONCopy(&result, spec); err != nil {
+		return
 	}
 
-	if err := compat.JSONCopy(&clusterMerged, spec); err != nil {
-		return false, err
+	// TODO: might need to do this with other slices
```

Comment:
can we file a ticket for this? Seems like we need to cover all slices with tests...